specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-06-20-preview/HybridCompute.json

```diff
@@ -1606,665 +1492,1589 @@
-        },
-        "detectedProperties": {
-          "$ref": "#/definitions/DetectedProperties",
-          "description": "Detected properties from the machine."
-        },
-        "networkProfile": {
-          "$ref": "#/definitions/NetworkProfile",
-          "description": "Information about the network the machine is on."
+          "description": "The URI to fetch the next page of Machines. Call ListNext() with this URI to fetch the next page of license profile."
         }
       },
-      "description": "Describes the properties of a hybrid machine."
+      "required": [
+        "value"
+      ],
+      "description": "The List hybrid machine license profile operation response."
     },
-    "MachineUpdateProperties": {
+    "EsuServerType": {
+      "type": "string",
+      "enum": [
+        "Standard",
+        "Datacenter"
+      ],
+      "x-ms-enum": {
+        "name": "EsuServerType",
+        "modelAsString": true
+      },
+      "description": "The server types for Esu."
+    },
+    "EsuEligibility": {
+      "type": "string",
+      "enum": [
+        "Eligible",
+        "Ineligible",
+        "Unknown"
+      ],
+      "x-ms-enum": {
+        "name": "EsuEligibility",
+        "modelAsString": true
+      },
+      "description": "The ESU eligibility."
+    },
+    "EsuKey": {
       "type": "object",
       "properties": {
-        "locationData": {
-          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/locationData"
-        },
-        "osProfile": {
-          "$ref": "#/definitions/OSProfile"
-        },
-        "cloudMetadata": {
-          "$ref": "#/definitions/CloudMetadata",
-          "description": "The metadata of the cloud environment (Azure/GCP/AWS/OCI...)."
-        },
-        "agentUpgrade": {
-          "$ref": "#/definitions/AgentUpgrade",
-          "description": "The info of the machine w.r.t Agent Upgrade"
-        },
-        "parentClusterResourceId": {
+        "sku": {
           "type": "string",
-          "description": "The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any."
+          "description": "SKU number."
         },
-        "privateLinkScopeResourceId": {
+        "licenseStatus": {
           "type": "string",
-          "description": "The resource id of the private link scope this machine is assigned to, if any."
+          "description": "The current status of the license profile key."
         }
       },
-      "description": "Describes the ARM updatable properties of a hybrid machine."
+      "description": "ESU key"
     },
-    "Machine": {
+    "EsuKeyState": {
+      "type": "string",
+      "enum": [
+        "Inactive",
+        "Active"
+      ],
+      "x-ms-enum": {
+        "name": "EsuKeyState",
+        "modelAsString": true
+      },
+      "description": "The ESU key state."
+    },
+    "LicenseProfileStorageModelEsuProperties": {
```

Comment:
the storage model representation isnt returned. we have an ARM model and a machineInstanceView model that we use

Reply:
LicenseProfileMachineInstanceViewEsuProperties is a type of LicenseProfileArmEsuProperties which is a type of LicenseProfileStorageModelEsuProperties... I modeled it the way we did it in the code base